.github/workflows/ci.yml

```diff
@@ -21,10 +21,10 @@ jobs:
     if: github.repository == 'scientific-python/upload-nightly-action'
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29  # v4.1.6
```

Comment:
@tupui you said that maybe we had to use hash:hash ?

Reply:
I'm slightly less concerned about these for projects that are known. But maybe I don't understand @tupui's point.

Reply:
I am trying to find where I read that 😅 Still digging but yes the hack was that if you had a branch named as the hash, then it would be picked up. In the article they were mentioning some protections from GitHub but there was still a way to do the hack IIRC.